controllers/dspipeline_params.go

```diff
@@ -372,8 +394,22 @@ func (p *DSPAParams) SetupObjectParams(ctx context.Context, dsp *dspa.DataScienc
 		}
 		p.ObjectStorageConnection.AccessKeyID = accessKey
 		p.ObjectStorageConnection.SecretAccessKey = secretKey
+
 	}
 
+	if p.ExternalRouteEnabled(dsp) {
+		route, err := p.RetrieveAndSetExternalRoute(ctx, client, log)
+		if err != nil {
+			log.Info("Unable to retrieve route", "error", err)
+		}
+		p.ObjectStorageConnection.ExternalRouteURL = route.Spec.Host
+		p.ObjectStorageConnection.Endpoint = route.Spec.Host
+		p.ObjectStorageConnection.Secure = util.BoolPointer(true)
+		p.ObjectStorageConnection.Host = route.Spec.Host
```

Comment:
accessing `route.` here looks like it's prone to error because it hasn't been created yet, have you looked into just formulating the route? it's always the same format: 

```
https://${route_name}-${namespace}.${api-server-hostname}
```

route_name/namespace you already know, just check if we can easily figure out the api-server hostname, then you don't need to do a get for a route here